src/cargo/core/compiler/unit_dependencies.rs

```diff
@@ -830,21 +807,44 @@ impl<'a, 'cfg> State<'a, 'cfg> {
             .unwrap_or_else(|_| panic!("expected {} to be downloaded", id))
     }
 
-    /// Returns a filtered set of dependencies for the given unit.
-    fn deps(
+    /// Returns non-custom and non-transitive dependencies.
+    fn non_custom_build_and_non_transitive_deps(
```

Comment:
Can you keep the method name the same?  It is understood that this method is already filtering some deps.